airflow/serialization/pydantic/dag_run.py

```diff
@@ -26,7 +26,7 @@
 class DagRunPydantic(BaseModelPydantic):
     """Serializable representation of the DagRun ORM SqlAlchemyModel used by internal API."""
 
-    id: int
+    id: Optional[int]
```

Comment:
This field should not be None

Reply:
I also thought that, but it is None during creation (before submission to DB) -  it is assigned by SQLAlchemy after you run 
```
session.add(obj)
session.commit()
```

Reply:
But when do we ever serialize an unsaved DagRun?

Reply:
I’m not sure Job should have an optional ID either. Maybe make this not optional for now and we can punt this until it’s nocessary for a model. It’s always easier to remove a restriction than add one.